src/main/kotlin/no/nav/syfo/dialogmote/DialogmoteService.kt

```diff
@@ -653,6 +650,107 @@ class DialogmoteService(
         }
     }
 
+    suspend fun endreFerdigstiltReferat(
+        callId: String,
+        dialogmote: Dialogmote,
+        opprettetAv: String,
+        referat: NewReferatDTO,
+        token: String,
+    ) {
+        if (dialogmote.status != DialogmoteStatus.FERDIGSTILT) {
+            throw ConflictException("Failed to Endre Ferdigstilt Dialogmote, not Ferdigstilt")
+        }
+
+        val virksomhetsnummer = dialogmote.arbeidsgiver.virksomhetsnummer
+
+        val narmesteLeder = narmesteLederClient.activeLeder(
+            personIdentNumber = dialogmote.arbeidstaker.personIdent,
+            virksomhetsnummer = virksomhetsnummer,
+            callId = callId
+        )
+
+        val pdfReferat = pdfGenClient.pdfReferat(
+            callId = callId,
+            documentComponentDTOList = referat.document,
+        ) ?: throw RuntimeException("Failed to request PDF - Referat")
+
+        val digitalVarsling = isDigitalVarselEnabled(
+            personIdentNumber = dialogmote.arbeidstaker.personIdent,
+            token = token,
+            callId = callId,
+        )
+
+        database.connection.use { connection ->
+
+            if (dialogmote.tildeltVeilederIdent != opprettetAv) {
+                connection.updateMoteTildeltVeileder(
+                    commit = false,
+                    moteId = dialogmote.id,
+                    veilederId = opprettetAv,
+                )
+            }
+            val (pdfId, _) = connection.createPdf(
+                commit = false,
+                pdf = pdfReferat,
+            )
+            val newReferat = referat.toNewReferat(
+                moteId = dialogmote.id,
+                ferdigstilt = true,
+            )
+            val existingReferat = dialogmote.referatList.firstOrNull()
+                ?: throw RuntimeException("Ferdigstilt mote ${dialogmote.id} does not have referat")
+
+            val (_, referatUuid) = if (existingReferat.ferdigstilt) {
```

Comment:
Kan man endre et referat som ikke er ferdigstilt?

Reply:
Hmja, riktig. For i det tilfellet vil `existingReferat` være et mellomlagret referat (som endrer et ferdigstilt referat). Mens ellers er `existingReferat` det referatet som skal endres. Må holde tunga rett i munnen her 🤯